infrastructure/http_metadata_service_test.go

```diff
@@ -422,4 +422,66 @@ func describeHTTPMetadataService() {
 		})
 
 	})
+
+	Describe("Decode a string of user data which has been encoded using base-64 encoding by IaaS", func() {
+		var (
+			ts *httptest.Server
+		)
+
+		createHandlerFunc := func(count int) func(http.ResponseWriter, *http.Request) {
+			initialCount := 0
```

Comment:
I had updated the test case with correct encoded string,  and at present, unit tests passed.